frontend/src/components/NavBar.js

```diff
@@ -37,6 +37,7 @@ import { executeCommand } from '../util/cli_utils';
 import DOMPurify from 'dompurify';
 
 const kDefaultWorksheetName = 'unnamed';
+const NAME_STR_REGEX = /^[a-zA-Z_][a-zA-Z0-9_.-]*$/; // matches regex on the backend in spec_util.py
```

Comment:
Can you rename this to NAME_REGEX, and also extract this regex into constants.js? It's also declared / used in NewWorksheet.js, so we want to only define it in one place.